src/Http/SetupRequest.php

```diff
@@ -31,4 +31,9 @@ public function redirectUrl(): ?string
     {
         return $this->input('data.redirect_url');
     }
+
+    protected function settingsRules(): string|array
+    {
+        return 'array';
+    }
```

Comment:
I know I suggested exactly this code but I'm wondering, isn't the following the same?
```suggestion
    protected function settingsRules(): array
    {
        return ['array'];
    }
```
Just so we avoid having to return string as well. Not sure if it will work!

Reply:
it will work, but we lose the flexibility to return, for example
```
'array|size:3'
```
Now that I'm thinking about it longer, I don't see how this helps us validate individual settings. In case of pay.nl, we want to be able to validate not just `data.settings`, but specifically `data.settings.api_code`, `data.settings.api_token` and `data.settings.service_id`. All those keys have to go directly into the array that is returned by `rules()`, it won't work with:
```
protected function settingsRules(): array
{
    return [
        'api_code' => 'required|string',
        ...
    ];
}
```